### PR TITLE
[WIP] feat: add typography

### PR DESCRIPTION
**Pull request by zkwolf**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch. Pull request will be merged after one of collaborators approve. Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/vueComponent/ant-design-vue/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

> 1. Describe the source of requirement.
> 2. Resolve what problem.
> 3. Related issue link.

### API Realization (Optional if not new feature)

> 1. Basic thought of solution and other optional proposal.
> 2. List final API realization and usage sample.
> 3. GIF or snapshot should be provided if includes UI/interactive modification.

### What's the effect? (Optional if not new feature)

> 1. Does this PR affect user? Which part will be affected?
> 2. What will say in changelog?
> 3. Does this PR contains potential break change or other risk?

### Changelog description (Optional if not new feature)

> 1. English description
> 2. Chinese description (optional)

### Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### Additional Plan? (Optional if not new feature)

Still something to do
- [ ] type definition
- [ ] lose some tese case
- [ ] rtl direction

> If this PR related with other PR or following info. You can type here.

### CODE REVIEW

File: components/typography/Base.jsx [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+  code: PropTypes.bool,
+  mark: PropTypes.bool,
+  underline: PropTypes.bool,
+  delete: PropTypes.bool,
+  strong: PropTypes.bool,
+};
+
+const ELLIPSIS_STR = '...';
+
+const Fragment = {
```
Comment (Amour1688) [inline components/typography/Base.jsx:42]: 这个建议抽到公共的`util`里面

File: components/typography/Base.jsx [hunk 2]
```diff
@@ -0,0 +1,450 @@
...
+      return getPrefixCls('typography', customizePrefixCls);
+    },
+    renderContent(locale) {
+      const { ellipsisContent, isEllipsis, expanded } = this;
+      const { component, type, disabled, style, title, ...restProps } = this.$props;
+      const { ['aria-label']: customAriaLabel } = this.$attrs;
+      const children = this.$slots.default;
+      const { rows, suffix } = this.getEllipsis();
+
+      this.editStr = locale.edit;
```
Comment (Amour1688) [inline components/typography/Base.jsx:353]: 解构？

File: components/typography/Editable.jsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+      inComposition: false,
+    };
+  },
+  watch: {
+    value(val) {
+      this.current = val;
+    },
+  },
+  mounted() {
+    const resizableTextArea = this.$refs.textarea.$refs.resizableTextArea;
```
Comment (Amour1688) [inline components/typography/Editable.jsx:26]: 需要判空一下吗

Comment (zkwolf) [reply components/typography/Editable.jsx:26]: 可以判一下，不过最新的textarea基于resizableTextArea，理论来说不会有空的可能

**Comment by zkwolf** [comment]
rebase to 1.x, close

**Comment by theseanharris1** [comment]
@zkwolf - would be incredible if you're able to push this 👍🏼

**Comment by zkwolf** [comment]
I will focus on this on the next week.

**Pull request closed without merging**